xapian-core/weight/lmweight.cc

```diff
@@ -229,29 +225,30 @@ LMWeight::get_maxpart() const
 
 /* The extra weight component in the Dir+ formula is :-
  *
- * |Q| + log (param_smoothing1 / |D| + param_smoothing1)
+ * |Q| * log (param_smoothing1 / (|D| + param_smoothing1))
  * 
  * where, |Q| is total query length.
  *	  |D| is total document length.
  */
 double
 LMWeight::get_sumextra(Xapian::termcount len, Xapian::termcount) const
 {
-    if (enable_dirplus) {
-	double len_double = len;
-	double extra_weight = param_smoothing1 / (len_double + param_smoothing1);
-	return get_query_length() * log (extra_weight);
-    } else
+    if (select_smoothing == DIRICHLET_PLUS_SMOOTHING) {
+	double extra_weight = param_smoothing1 / (len + param_smoothing1);
+	return get_query_length() * log(extra_weight);
+    }
+    else
 	return 0;
 }
```

Comment:
The layout of the `else` clause here doesn't match our coding standards (it should be `} else` on one line.

But in a case like this where you've just handled a particular case with an `if` that ends in a `return` statement, and there's a default return following it, it's probably more readable just to remove the `else` and have the `return 0;` at the top level of the function:

```
{
    if (select_smoothing == DIRICHLET_PLUS_SMOOTHING) {
        double extra_weight = param_smoothing1 / (len + param_smoothing1);
        return get_query_length() * log(extra_weight);
    }

    return 0;
}
```